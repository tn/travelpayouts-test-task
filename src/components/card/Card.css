@import '../../Config.css';

.card {
    display: flex;
    flex-flow: row;
    align-items: flex-end;
    justify-content: flex-end;
    max-width: 1136px;
    padding: 26px 32px;
    margin-bottom: 24px;
    background-color: #fff;
    border: 1px solid #E2E5EC;
    border-radius: 6px;
}

.card__info {
    margin-right: auto;
}

.card__title {
    margin: 0 0 5px;
    font-size: 29px;
    font-weight: 300;
    line-height: 42px;
}

.card__desc {
    font-size: 15px;
    line-height: 22px;
    color: var(--secondaryTextColor);
}

.card__action {
    position: relative;
    margin-right: 24px;
}

.card__copy {
    position: absolute;
    right: 8px;
    top: 34px;
    font-size: 0;
    text-indent: -9999px;
    border: 0;
    margin: 0;
    width: 20px;
    height: 20px;
    padding: 15px;
    background: url('./copy.svg') center center no-repeat;
    cursor: pointer;
    opacity: 0.8;
}

.card__copy:hover {
    opacity: 1;
}
