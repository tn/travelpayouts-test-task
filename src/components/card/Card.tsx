import * as React from 'react'
import './Card.css'
import Button from '../button/Button'
import Input from '../input/Input'

const Card: React.SFC = () => (
  <div className='card'>
    <div className='card__info'>
      <h3 className='card__title'>Name</h3>
      <div className='card__desc'>Desc</div>
    </div>
    <div className='card__action'>
      <Input labelId='promocode' value='Travelpayouts' readOnly={true}>Промокод</Input>
      <button className='card__copy'>Copy</button>
    </div>
    <Button primary={true}>Получить бонус</Button>
  </div>
)

export default Card
