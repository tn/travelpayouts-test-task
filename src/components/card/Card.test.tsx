import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Card from './Card'

configure({ adapter: new Adapter() })

describe('Card', () => {
  const props = {
    title: 'Some title',
    desc: 'Description',
    link: 'http://example.com',
    promo: 'travelpayouts'
  }

  it('should be rendered', () => {
    const wrapper = renderer.create(<Card {...props} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
