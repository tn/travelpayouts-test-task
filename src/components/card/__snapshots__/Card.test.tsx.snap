// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be rendered 1`] = `
<div
  className="card"
>
  <div
    className="card__info"
  >
    <h3
      className="card__title"
    >
      <a
        className="card__link"
        href="http://example.com"
        rel="noopener nofollow"
        target="_blank"
      >
        Some title
      </a>
    </h3>
    <div
      className="card__desc"
    >
      Description
    </div>
  </div>
  <div
    className="card__action"
  >
    <label
      className="field__label"
      htmlFor="promocode"
    >
      Промокод 
      
    </label>
    <input
      className="field"
      id="promocode"
      readOnly={true}
      value="travelpayouts"
    />
    <button
      className="card__copy"
      onClick={[Function]}
    >
      Copy
    </button>
  </div>
  <button
    className="button button--primary"
  >
    Получить бонус
  </button>
</div>
`;
